mantidproject/mantid
# Change the name of the output workspaces in SANSILLAutoProcess

**Description of work.**

This PR changes the naming convention of the output workspaces of SANSILLAutoProcess. The detector distance, the collimation distance and the wavelength are appended to the `OutputWorkspace` parameter value.

This applies for:
* the sample output workspaces
* the panels output workspaces (`OutputPanels == true`)
* the wedges output workspaces (`OutputType == I(Q) && NumberOfWedges != 0`)

**To test:**

With data from `SystemTest/ILL/D11/`
```python
SANSILLAutoProcess(SampleRuns="2889,2885,2881", OutputWorkspace="ws")
```
Check that you obtain a group workspace named `ws` that contains 1 workspace per distance and a stitched one. Check that the values in the names of the individual reduction correspond to the sample logs value found in `002889.nxs`, `002885.nxs` and `002881.nxs`:
* `dX.Xm`: detector distance. `L2` key in the sample logs
* `cX.Xm`: collimation position. `collimation.actual_position` key of the sample logs
* `wX.XA`: wavelength. `wavelength` key (or `selector.wavelength` if `wavelength` does not exist) of the sample logs

---

```python
SANSILLAutoProcess(SampleRuns="2889,2885,2881", OutputWorkspace="ws", NumberOfWedges=2)
```
Idem for the sample output workspaces. Check that you also obtain two other group workspaces `ws_wedge_1` and `ws_wedge_2` which contain workspaces with names following the same scheme.

---

With data from `UnitTest/ILL/D33/`:
```python
SANSILLAutoProcess(SampleRuns="001464", OutputWorkspace="ws", OutputPanels=True)
```
Idem but with a `ws_panels` group workspace. In this case, the panel distance is used.


Part of #30343 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
